packages/cursorless-engine/src/languages/TreeSitterQuery/queryPredicateOperators.ts

```diff
@@ -73,6 +73,22 @@ class HasMultipleChildrenOfType extends QueryPredicateOperator<HasMultipleChildr
   }
 }
 
+/**
+ * A predicate operator that returns true if the node has more than 1 child not of
+ * type {@link type} (inclusive).  For example, `(has-multiple-children-not-of-type?
+ * @foo bar)` will accept the match if the `@foo` capture has 2 or more children
+ * not of type `bar`.
+ */
+class HasMultipleChildrenNotOfType extends QueryPredicateOperator<HasMultipleChildrenNotOfType> {
```

Comment:
Wow this really feels like a slippery slope 😅. Getting to the point where I wonder if we want to create our own mini-language

Reply:
dsl ftw! :D